src/replay_stage.rs

```diff
@@ -223,13 +215,12 @@ impl ReplayStage {
 
         // Update Tpu and other fullnode components with the current bank
         let (mut current_slot, mut current_leader_id, mut max_tick_height_for_slot) = {
-            let leader_scheduler = leader_scheduler.read().unwrap();
-            let slot = leader_scheduler.tick_height_to_slot(tick_height + 1);
+            let slot = (tick_height + 1) / bank.ticks_per_slot();
+            let epoch = slot / bank.slots_per_epoch();
             let first_tick_in_slot = slot * bank.ticks_per_slot();
 
-            let leader_id = leader_scheduler
-                .get_leader_for_slot(slot)
-                .expect("Leader not known after processing bank");
+            let leader_id =
+                LeaderScheduler1::default().slot_leader_by(&bank, |_, _, _| (slot, epoch));
```

Comment:
That `tick_height + 1` is bizarre. Is the intent similar to `next_slot_leader()`?

Reply:
I think intent is to find if the next tick might belong to the next slot. It may still belong to the current slot. In that case the leader will not change.

Reply:
weird

Reply:
Let's not worry about these edge cases in this PR. Instead, just document them in a new issue and let's get this one merged once you have the test suite passing. So much wonderful deleting here!

Reply:
a slot-based replay stage kills all the +1

Reply:
👍